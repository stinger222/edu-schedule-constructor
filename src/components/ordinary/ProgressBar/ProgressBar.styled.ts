import styled from "styled-components";

export const StyledProgressBar = styled.div`
	display: flex;
	width: 6.5em;

	& .caption {
		height: 100%;
		width: 75%;
		font-size: 1.5em;
		margin-right: 0.4em;
	}

	& .caption > div {
		text-align: center;
		font-weight: 600;
	}

	& .caption .caption_end {
		font-size: 0.85em;
		color: gray
	}

	& .progress_bar_body {
		height: 100%;
		width: 25%;
	}

	& .progress_bar_body .indicator {
		width: 1em;
		height: 1em;
		background: transparent;
		box-shadow: 0 0 0 0.15em #0075FF inset;
		border-radius: 5em;
		margin-inline: auto;
		margin-bottom: 0.4em;
	}

	&.active .progress_bar_body .indicator {
		height: 1.3em;
		width: 1.3em;
		position: relative;
		box-shadow: 0 0 0 0.13em #0075FF inset;
	}

	&.active .progress_bar_body .indicator:after {
		content: "";
		position: absolute;
		inset: 50%;
		transform: translate(-50%, -50%);
		height: 0.4em;
		width: 0.4em;
		box-shadow: 0 0 0 0.113em #0075FF;
		/* box-shadow: 0 0 0 0.2em red; */
		border-radius: 10em;
		background: #0075ff;
	}
	
	& .progress_bar_body .line {
		width: 0.25em;
		height: calc(100% - 0.8em);
		border-radius: 1em;
		background: #0075FF;
		margin-inline: auto;
	}
`

