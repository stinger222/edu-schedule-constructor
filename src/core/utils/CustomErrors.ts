export class EmptyDay extends Error {
  readonly name = "EmptyDay"

  constructor() {
    super()
  }
}

export class Warning extends Error {
  readonly name = "Warning"

  constructor(message: string) {
    super(message)
  }
}
