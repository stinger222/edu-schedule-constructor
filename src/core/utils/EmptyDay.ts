class EmptyDay extends Error {
  name = "EmptyDay"

  constructor(message: string) {
    super(message)
  }
}

export default EmptyDay
