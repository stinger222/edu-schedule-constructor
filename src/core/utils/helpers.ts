import { weekDates, Replacements, Cases } from './../types/types';

export const getCurrentWeekDates = (): weekDates => {
	const result = []
	const date = new Date()
	date.setDate(date.getDate() - (date.getDay() === 0 ? 7 : date.getDay())) // now it's current week sunday date
	for (let i = 0; i < 7; i++) {
		date.setDate(date.getDate() + 1) // and now it's monday
		result.push(date.getDate())
	}

	return formatNumbers(result) as weekDates
}

export const formatNumbers = (numbers: number[]): string[] => {
	return numbers.map(number => formatNumber(number))
}

export function replaceBlankProps<P extends {[key: string]: any}> (props: P , replacements: Replacements<P>): P {
	const result: any = {}
	for (const [key, value] of Object.entries(props)) {
		if (typeof value === 'string') {
			if (value.trim() == '') {
				result[key] = replacements[key as keyof typeof replacements]
			} else {
				result[key] = value
			}
		} else {
			result[key] = value
		}
	}
	return result as P
}

export const formatNumber = (number: number): string => {
	if (isNaN(number)) return "??"
	
	return Math.abs(number).toLocaleString('en-US', {
    minimumIntegerDigits: 2,
    useGrouping: false
  })
}

export const formatTimeString = (timeString: string): string => {
	let [hours, minutes] = timeString.split(':')

	return `${formatNumber(parseInt(hours))}:${formatNumber(parseInt(minutes))}`
}

export class WeekDays {
	private static _week = {
		short: ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вс'],
		full: {
			nominative: [
				'Понедельник', 'Вторник', 'Среда', 'Четверг', 'Пятница', 'Суббота', 'Воскресенье'
			],
			accusative: [
				'Понедельник', 'Вторник', 'Среду', 'Четверг', 'Пятницу', 'Субботу', 'Воскресенье'
			]
		} as Record<Cases, string[]>
	}

	static getShort() {
		return this._week.short
	}

	static getFull(_case: Cases = Cases.Nominative, lowerCase: boolean = false) {
		let result = this._week.full[_case]
		return lowerCase ? result.map(i => i.toLowerCase()) : result
	}
}