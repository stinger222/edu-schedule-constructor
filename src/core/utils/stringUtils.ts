export const formatNumbers = (numbers: number[]): string[] => {
	return numbers.map(number => formatNumber(number))
}

export const formatNumber = (number: number, digits: number = 2): string => {
	if (isNaN(number)) return "??"
  
	const formatter = new Intl.NumberFormat("en-US", {
		minimumIntegerDigits: digits,
		useGrouping: true
	})
  
	return formatter.format(Math.abs(number))
}

export const formatTimeString = (timeString: string): string => {
	const [hours, minutes] = timeString.split(":")

	return `${formatNumber(parseInt(hours))}:${formatNumber(parseInt(minutes))}`
}

export const capitalize = (str: string, eachWord: boolean = false): string => {
	if (eachWord) {
		return str
			.split(" ")
			.map(s => s.charAt(0).toUpperCase() + s.slice(1))
			.join(" ")
	}
	return str.charAt(0).toUpperCase() + str.slice(1)
}

export const validateField = (value: string) => {
	value = value.trim()
	return value !== "" && value !== "undefined"
}