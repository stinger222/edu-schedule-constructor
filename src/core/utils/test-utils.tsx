import { ReactElement } from "react"
import { DarkTheme } from "../themes/Dark"
import { ThemeProvider } from "styled-components"
import { render, RenderOptions } from "@testing-library/react"
import GlobalStyles from "../themes/GlobalStyles"

const AllTheProviders: React.FC<{children: ReactElement}> = ({children}) => {
	return (
		<ThemeProvider theme={DarkTheme}>
			{children}
		</ThemeProvider>
	)
}

const myRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'queries'>) => {
	return render(ui, { wrapper: AllTheProviders, ...options,})
}

export * from '@testing-library/react'
export {myRender as render}