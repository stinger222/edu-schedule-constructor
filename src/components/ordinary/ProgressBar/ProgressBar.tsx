import React from "react"
import { Replacements } from "../../../core/types/types"
import { StyledProgressBar } from "./ProgressBar.styled"
import { formatTimeString, replaceBlankProps } from "../../../core/utils/helpers"
import { boolean } from "yup"

interface IProps {
	startTime: string,
	endTime: string,
	active?: boolean
}

const propsReplacements: Replacements<IProps> = {
	startTime: "??:??",
	endTime: "??:??",
}

const ProgressBar: React.FC<IProps> = (props) => {
	let { startTime, endTime, active } = replaceBlankProps<IProps>(props, propsReplacements)

	startTime = formatTimeString(startTime)
	endTime = formatTimeString(endTime)

	const validValue = /^[0-9]{2}:[0-9]{2}$/
	if (validValue.test(startTime) && validValue.test(endTime)) {
		if (new Date(`1/1/2000 ${startTime}`) > new Date(`1/1/2000 ${endTime}`)) { // if startTime bigger that endTime...
			[startTime, endTime] = [endTime, startTime] // ...swap them
		}
	}

	return (
		<StyledProgressBar className={active ? 'active' : ''}>
			<div className="caption">
				<div className="caption_start">{ startTime }</div>
				<div className="caption_end">{ endTime }</div>
			</div>
			<div className="progress_bar_body">
				<div className="indicator"></div>
				<div className="line"></div>
			</div>
		</StyledProgressBar>
	)
}

export default React.memo(ProgressBar)
