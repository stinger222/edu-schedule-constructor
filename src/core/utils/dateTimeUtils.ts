import { Cases, weekDates } from "../types/types"
import { formatNumber } from "./stringUtils"


interface IData {
  ru: {
    short: readonly string[],
    full: {
      nominative: readonly string[],
      accusative: readonly string[]
    }
  },
  en: {
    short: readonly string[],
    full: {
      nominative: readonly string[],
      accusative: readonly string[]
    }
  }
}
export class WeekDays {

  private static readonly data: IData = {
    ru: {
      short: ["Пн", "Вт", "Ср", "Чт", "Пт", "Сб", "Вс"],
      full: {
        nominative: [
          "Понедельник", "Вторник", "Среда", "Четверг", "Пятница", "Суббота", "Воскресенье"
        ],
        accusative: [
          "Понедельник", "Вторник", "Среду", "Четверг", "Пятницу", "Субботу", "Воскресенье"
        ]
      }
    },
    en: {
      short: ["M", "T", "W", "T", "F", "S", "S"],
      full: {
        nominative: [
          "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"
        ],
        accusative: [
          "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"
        ]
      }
    }
  }

	static getShort(lang: "ru" | "en"): ReadonlyArray<string> {
		return this.data[lang].short as ReadonlyArray<string>
	}

	static getFull(lang: "ru" | "en", caseType: Cases = Cases.Nominative, toLowerCase: boolean = false): ReadonlyArray<string> {
		const result: ReadonlyArray<string> = this.data[lang].full[caseType]
		return toLowerCase ? result.map(i => i.toLowerCase()) : result
	}
}

export const getCurrentWeekDates = (): weekDates => {
	const date = new Date()
	date.setDate(date.getDate() - (date.getDay() === 0 ? 7 : date.getDay())) // now it's current week sunday date:

	const result = Array.from({ length: 7 }, () => {
		date.setDate(date.getDate() + 1) // and now it's monday
		return formatNumber(date.getDate())
	})

	return result as weekDates
}
