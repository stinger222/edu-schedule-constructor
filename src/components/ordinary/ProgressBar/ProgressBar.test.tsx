import { cleanup, render, screen } from "../../../core/utils/test-utils"
import ProgressBar from "./ProgressBar"

describe('Testing ProgressBar render with different props', () => {
	it("Renders ProgressBar with valid props", () => {
		const progressBar = render(<ProgressBar startTime="10:00" endTime="11:00"/>)

		expect(screen.getByText("10:00", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("11:00", {selector: ".caption-end"})).toBeInTheDocument()
		expect(progressBar).toMatchSnapshot()
	})

	it("Renders ProgressBar correctly with logically incorrect props", () => {
		const progressBar = render(<ProgressBar startTime="10:00" endTime="09:00"/>)

		expect(screen.getByText("09:00", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("10:00", {selector: ".caption-end"})).toBeInTheDocument()
		expect(progressBar).toMatchSnapshot()

		cleanup()

		render(<ProgressBar startTime="05:30" endTime="03:20"/>)

		expect(screen.getByText("03:20", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("05:30", {selector: ".caption-end"})).toBeInTheDocument()
		
		render(<ProgressBar startTime="08:01" endTime="08:00"/>)

		expect(screen.getByText("08:00", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("08:01", {selector: ".caption-end"})).toBeInTheDocument()
	})

	it("Renders ProgressBar with valid props, but values not formatted", () => {
		const progressBar = render(<ProgressBar startTime="7:3" endTime="9:5"/>)

		expect(screen.getByText("07:03", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("09:05", {selector: ".caption-end"})).toBeInTheDocument()
		expect(progressBar).toMatchSnapshot()

		cleanup()

		render(<ProgressBar startTime="0:0" endTime="4:5"/>)

		expect(screen.getByText("00:00", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("04:05", {selector: ".caption-end"})).toBeInTheDocument()
	})

	it("Renders ProgressBar with string props that filled with spaces", () => {
		render(<ProgressBar startTime="  " endTime="    "/>)

		expect(screen.getByText("??:??", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("??:??", {selector: ".caption-end"})).toBeInTheDocument()

		cleanup()

		render(<ProgressBar startTime="10:00" endTime="    "/>)

		expect(screen.getByText("10:00", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("??:??", {selector: ".caption-end"})).toBeInTheDocument()
		
		cleanup()

		render(<ProgressBar startTime=" " endTime="09:00"/>)

		expect(screen.getByText("??:??", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("09:00", {selector: ".caption-end"})).toBeInTheDocument()
	})

	it("Renders ProgressBar with epmpty string props", () => {
		render(<ProgressBar startTime="" endTime=""/>)

		expect(screen.getByText("??:??", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("??:??", {selector: ".caption-end"})).toBeInTheDocument()

		cleanup()

		render(<ProgressBar startTime="09:00" endTime=""/>)

		expect(screen.getByText("09:00", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("??:??", {selector: ".caption-end"})).toBeInTheDocument()

		cleanup()

		render(<ProgressBar startTime="" endTime="10:00"/>)

		expect(screen.getByText("??:??", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("10:00", {selector: ".caption-end"})).toBeInTheDocument()
	})

	it("Renders ProgressBar with invalid props", () => {
		const progressBar = render(<ProgressBar startTime="cum" endTime="sock"/>)
		
		expect(screen.getByText("??:??", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("??:??", {selector: ".caption-end"})).toBeInTheDocument()
		expect(progressBar).toMatchSnapshot()

		cleanup()

		render(<ProgressBar startTime="08:30" endTime="sock"/>)
		
		expect(screen.getByText("08:30", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("??:??", {selector: ".caption-end"})).toBeInTheDocument()

		cleanup()

		render(<ProgressBar startTime="cum" endTime="9:40"/>)
		
		expect(screen.getByText("??:??", {selector: ".caption-start"})).toBeInTheDocument()
		expect(screen.getByText("09:40", {selector: ".caption-end"})).toBeInTheDocument()
	})
})
