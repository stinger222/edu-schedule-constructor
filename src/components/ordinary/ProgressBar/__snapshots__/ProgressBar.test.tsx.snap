// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ProgressBar render with different props Renders ProgressBar correctly with logically incorrect props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe hRtsXN"
      >
        <div
          class="caption"
        >
          <div
            class="caption-start"
          >
            09:00
          </div>
          <div
            class="caption-end"
          >
            10:00
          </div>
        </div>
        <div
          class="progress-bar-body"
        >
          <div
            class="indicator"
          />
          <div
            class="line"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe hRtsXN"
    >
      <div
        class="caption"
      >
        <div
          class="caption-start"
        >
          09:00
        </div>
        <div
          class="caption-end"
        >
          10:00
        </div>
      </div>
      <div
        class="progress-bar-body"
      >
        <div
          class="indicator"
        />
        <div
          class="line"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing ProgressBar render with different props Renders ProgressBar with invalid props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe hRtsXN"
      >
        <div
          class="caption"
        >
          <div
            class="caption-start"
          >
            ??:??
          </div>
          <div
            class="caption-end"
          >
            ??:??
          </div>
        </div>
        <div
          class="progress-bar-body"
        >
          <div
            class="indicator"
          />
          <div
            class="line"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe hRtsXN"
    >
      <div
        class="caption"
      >
        <div
          class="caption-start"
        >
          ??:??
        </div>
        <div
          class="caption-end"
        >
          ??:??
        </div>
      </div>
      <div
        class="progress-bar-body"
      >
        <div
          class="indicator"
        />
        <div
          class="line"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing ProgressBar render with different props Renders ProgressBar with valid props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe hRtsXN"
      >
        <div
          class="caption"
        >
          <div
            class="caption-start"
          >
            10:00
          </div>
          <div
            class="caption-end"
          >
            11:00
          </div>
        </div>
        <div
          class="progress-bar-body"
        >
          <div
            class="indicator"
          />
          <div
            class="line"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe hRtsXN"
    >
      <div
        class="caption"
      >
        <div
          class="caption-start"
        >
          10:00
        </div>
        <div
          class="caption-end"
        >
          11:00
        </div>
      </div>
      <div
        class="progress-bar-body"
      >
        <div
          class="indicator"
        />
        <div
          class="line"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing ProgressBar render with different props Renders ProgressBar with valid props, but values not formatted 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe hRtsXN"
      >
        <div
          class="caption"
        >
          <div
            class="caption-start"
          >
            07:03
          </div>
          <div
            class="caption-end"
          >
            09:05
          </div>
        </div>
        <div
          class="progress-bar-body"
        >
          <div
            class="indicator"
          />
          <div
            class="line"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe hRtsXN"
    >
      <div
        class="caption"
      >
        <div
          class="caption-start"
        >
          07:03
        </div>
        <div
          class="caption-end"
        >
          09:05
        </div>
      </div>
      <div
        class="progress-bar-body"
      >
        <div
          class="indicator"
        />
        <div
          class="line"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
